aeon/anomaly_detection/_rockad.py

```diff
@@ -0,0 +1,274 @@
+    power_transformer_ : PowerTransformer
+        Transformer used to apply power transformation to the features.
+    n_inf_cols_ : list
+        List of feature columns containing infinite values, identified during fitting.
+    """
+
+    _tags = {
+        "capability:univariate": True,
+        "capability:multivariate": True,
+        "capability:missing_values": False,
+        "capability:multithreading": True,
+        "fit_is_empty": False,
+    }
+
+    def __init__(
+        self,
+        n_estimators=10,
+        n_kernels=100,
+        normalise=False,
+        n_neighbors=5,
+        n_jobs=1,
+        metric="euclidean",
+        power_transform=True,
+        window_size: int = 10,
+        stride: int = 1,
+        random_state=42,
+    ):
+        super().__init__(axis=0)
+
+        self.n_estimators = n_estimators
+        self.n_kernels = n_kernels
+        self.normalise = normalise
+        self.n_neighbors = n_neighbors
+        self.n_jobs = n_jobs
+        self.metric = metric
+        self.power_transform = power_transform
+        self.window_size = window_size
+        self.stride = stride
+        self.random_state = random_state
+
+        self.rocket_transformer_: Optional[Rocket] = None
+        self.estimator_: Optional[NearestNeighbors] = None
+        self.power_transformer_: Optional[PowerTransformer] = None
+        self.inf_columns_: Optional[np.ndarray] = None
+
+        if power_transform:
+            self.power_transformer_ = PowerTransformer()
+
+    def _fit(self, X: np.ndarray, y: Optional[np.ndarray] = None) -> "ROCKAD":
+        self._check_params(X)
+
+        _X, _ = sliding_windows(
+            X, window_size=self.window_size, stride=self.stride, axis=0
+        )
+
+        self._inner_fit(_X)
+
+        return self
+
+    def _check_params(self, X: np.ndarray) -> None:
+        if self.window_size < 1 or self.window_size > X.shape[0]:
+            raise ValueError(
+                "The window size must be at least 1 and at most the length of the "
+                "time series."
+            )
+
+        if self.stride < 1 or self.stride > self.window_size:
+            raise ValueError(
+                "The stride must be at least 1 and at most the window size."
+            )
+
+    def _inner_fit(self, X: np.ndarray) -> None:
+
+        if X.shape[0] < self.n_neighbors:
+            raise ValueError(
+                f"Window count ({X.shape[0]}) has to be larger than "
+                f"n_neighbors ({self.n_neighbors})."
+                "Please choose a smaller n_neighbors value or increase window count "
+                "by choosing a smaller window size or larger stride."
+            )
+
+        self.rocket_transformer_ = Rocket(
+            n_kernels=self.n_kernels,
+            normalise=self.normalise,
+            n_jobs=self.n_jobs,
+            random_state=self.random_state,
+        )
+
+        Xt = self.rocket_transformer_.fit_transform(X)
```

Comment:
It is really difficult to track shapes. Am I right with this?

```suggestion
        # X: (n_windows, window_size)
        Xt = self.rocket_transformer_.fit_transform(X)
        # XT: (n_cases, n_kernels)
```

Reply:
I think XT should be: (n_windows, n_kernels*2), Rocket producer for each feature 2 values but the windows remain. Or is n_cases just another term for window?

Reply:
Ok, `(n_windows, n_kernels*2)` makes sense, and `n_cases` = `n_windows` 👍🏼